include/WCSimWCDigitizerNew.hh

```diff
@@ -0,0 +1,107 @@
+#include "WCSimWCDAQMessenger.hh"
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimEnumerations.hh"
+#include "G4VDigitizerModule.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "globals.hh"
+#include "Randomize.hh"
+#include <map>
+#include <vector>
+#include <utility> //for std::pair
+
+
+// *******************************************
+// BASE CLASS
+// *******************************************
+
+class WCSimWCDigitizerBase : public G4VDigitizerModule
+{
+public:
+  
+  WCSimWCDigitizerBase(G4String name, WCSimDetectorConstruction*, WCSimWCDAQMessenger*, DigitizerType_t);
+  virtual ~WCSimWCDigitizerBase();
+  
+  bool AddNewDigit(int tube, int gate, float digihittime, float peSmeared, std::vector< std::pair<int,int> > digi_comp);
+  virtual void DigitizeHits(WCSimWCDigitsCollection* WCHCPMT) = 0;
+  void DigitizeGate(WCSimWCDigitsCollection* WCHC,G4int G);
+  void Digitize();
+
+  //.mac file option setting methods
+  void SetDigitizerDeadTime         (int deadtime) { DigitizerDeadTime = deadtime;         }; ///< Override the default digitizer deadtime (ns)
+  void SetDigitizerIntegrationWindow(int inttime ) { DigitizerIntegrationWindow = inttime; }; ///< Override the default digitizer integration window (ns)
+
+protected:
+  void ReInitialize() { DigiStoreHitMap.clear(); }
+
+  G4double peSmeared;
+
+  WCSimDetectorConstruction* myDetector; ///< Get the geometry information
+  WCSimWCDAQMessenger* DAQMessenger;     ///< Get the /DAQ/ .mac options
+
+  WCSimWCDigitsCollection*  DigiStore;
+  std::map<int,int> DigiStoreHitMap;   ///< Used to check if a digit has already been created on a PMT
+
+  //generic digitizer properties. Defaults set with the GetDefault*() methods. Overidden by .mac options
+  int DigitizerDeadTime;          ///< Digitizer deadtime (ns)
+  int DigitizerIntegrationWindow; ///< Digitizer integration window (ns)
+
+  DigitizerType_t DigitizerType; ///< Enumeration to say which digitizer we've constructed
+
+  virtual int GetDefaultDeadTime() = 0;          ///< Set the default digitizer-specific deadtime (in ns) (overridden by .mac)
+  virtual int GetDefaultIntegrationWindow() = 0; ///< Set the default digitizer-specific integration window (in ns) (overridden by .mac)
+
+  void GetVariables(); ///< Get the default deadtime, etc. from the derived class, and override with read from the .mac file
+};
+
+
+
+// *******************************************
+// DERIVED CLASSES
+// *******************************************
+
+
+// SKI-based digitizer class
+class WCSimWCDigitizerSKI : public WCSimWCDigitizerBase 
+{
+public:
+  
+  WCSimWCDigitizerSKI(G4String name, WCSimDetectorConstruction*, WCSimWCDAQMessenger*);
+  ~WCSimWCDigitizerSKI();
+
+  void DigitizeHits(WCSimWCDigitsCollection* WCHCPMT);
+
+private:
+  int GetDefaultDeadTime()          { return 0; }   ///< SKI digitizer deadtime is 0 ns
+  int GetDefaultIntegrationWindow() { return 400; } ///< SKI digitizer integration window is 400 ns
```

Comment:
Thinking about this a bit more since our meeting this morning, changing the default digitizer integration time should be it's own separate pull request. We should try and minimize small changes sneaking in with this issue so that people using the code aren't confused about why their output is suddenly different. Make an issue on GitHub, explain why the digitizer integration should be 400 instead of 200. Then, it should be a straightforward, one-line PR to submit and approve after this PR is merged.